core/src/main/java/hudson/model/RunParameterValue.java

```diff
@@ -88,9 +94,41 @@ public String getNumber() {
     	return r == null ? null : r[1];
     }
 
+    public class SerializableValue implements Serializable {
+
+        @Nonnull
+        public final String jobName;
+        public final int number;
+        @Nullable
+        public final String url;
+        @Nullable
+        public final String displayName;
```

Comment:
Why would you want to serialize it. It can be always restored on-demand, and this is a recommended way (e.g. Build Name Setter may change the name, etc.)